docs/advanced.md

```diff
@@ -2,50 +2,89 @@
 
 ## Client Instances
 
-Using a Client instance to make requests will give you HTTP connection pooling,
-will provide cookie persistence, and allows you to apply configuration across
-all outgoing requests.
-
 !!! hint
-    A Client instance is equivalent to a Session instance in `requests`.
+    If you are coming from Requests, `httpx.Client()` is what you can use instead of `requests.Session()`.
 
-!!! note
-    Starting from `httpx==0.10.0`, the default and recommended Client class is `AsyncClient`. This should help prevent breaking changes once sync support is reintroduced.
+### Why use a Client?
+
+!!! note "TL;DR"
+    If you do anything more than experimentation, one-off scripts, or prototypes, then you should use a `Client` instance.
+
+#### More efficient usage of network resources
+
+Using `Client` instances instead of the top-level API (`httpx.get()`, `httpx.post()`, etc.) generally results in **far more efficient usage of network resources** (i.e. HTTP connections).
+
+Benefits include:
+
+- Reduced latency across requests.
+- Reduced CPU usage.
+- Reduced bandwidth usage and network congestion.
+
+Indeed, when using the top-level API, an HTTP connection has to be opened and closed _for every single request_.
+
+So, if you are making 50 HTTP/1.1 requests per second to a host using the top-level API, then HTTPX needs to open and shutdown 50 connections per second. This is a lot of work! It can quickly become expensive, and slow down your programs. Reasons include higher latency due to handshaking, higher CPU usage due to repeated TLS handshakes, higher network congestion due to a higher number of TCP connections, etc.
```

Comment:
I would use this paragraph in place of the above benefits and maybe use bulletpoints in the reasons.

Reply:
Yes, the bullets and the "Reasons include" here do essentially repeat the same thing, so +1  to keeping just one of them, and using bullets.

Reply:
Thanks! I gave this section a different wording altogether to reduce the repetition. Let me know what you think!